code/modules/assembly/assembly.dm

```diff
@@ -3,6 +3,15 @@
 #define VALUE_VARIABLE_MIN  3 //Minimum possible number value
 #define VALUE_VARIABLE_MAX  4 //Maximum possible number value
 
+#define VALUE_IS_NUMBER(value_parameters) (value_parameters[VALUE_VARIABLE_TYPE] == VT_NUMBER || value_parameters[VALUE_VARIABLE_TYPE] == VT_POINTER)
```

Comment:
Pointers refer to an assembly's index in the assembly frame, so they're numbers. You can set them to a different number to make the assembly connect to something else